<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CustomMusicSetup</class>
 <widget class="QDockWidget" name="CustomMusicSetup">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>447</width>
    <height>576</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Music setup</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="setupZone">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tabMidi">
       <attribute name="title">
        <string notr="true">MIDI</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Synthesizer type:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1" colspan="3">
         <widget class="QComboBox" name="midiSynth"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="midiGainLabel">
          <property name="text">
           <string>Gain:</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSlider" name="midiGain">
          <property name="minimumSize">
           <size>
            <width>139</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximum">
           <number>400</number>
          </property>
          <property name="value">
           <number>200</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksAbove</enum>
          </property>
          <property name="tickInterval">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QDoubleSpinBox" name="midiGainAbs">
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>2.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QPushButton" name="midiGainReset">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="midiTempoLabel">
          <property name="text">
           <string>Tempo:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSlider" name="midiTempo">
          <property name="minimumSize">
           <size>
            <width>139</width>
            <height>0</height>
           </size>
          </property>
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>200</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksAbove</enum>
          </property>
          <property name="tickInterval">
           <number>25</number>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QDoubleSpinBox" name="midiTempoAbs">
          <property name="minimum">
           <double>0.010000000000000</double>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="3">
         <widget class="QPushButton" name="midiTempoReset">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="4">
         <widget class="QGroupBox" name="midiSetupADL">
          <property name="title">
           <string>Extra settings</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2" columnstretch="0,1000,1">
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <item row="1" column="0">
            <widget class="QLabel" name="label_9">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Custom bank:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Volume model:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="0" colspan="2">
            <widget class="QCheckBox" name="midiExAdlDeepVibrato">
             <property name="text">
              <string>Deep vibrato</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
             <property name="tristate">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string comment="The OPL3 timbre/instruments bank to change the MIDI music sounding.">Bank:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0" colspan="2">
            <widget class="QCheckBox" name="midiExAdlDeepTremolo">
             <property name="text">
              <string>Deep tremolo</string>
             </property>
             <property name="checked">
              <bool>false</bool>
             </property>
             <property name="tristate">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="midiExAdlCustomBank">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string notr="true">&lt;WIP&gt;</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="midiExAdlBrowsebank">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Browse...</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1" colspan="2">
            <widget class="QComboBox" name="midiExAdlBank">
             <property name="toolTip">
              <string>The timbre bank declares the sounding of MIDI music. Use different banks to compare the sounding difference.</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1" colspan="2">
            <widget class="QComboBox" name="midiExAdlVolumeModel">
             <property name="toolTip">
              <string>Volume model declares how music volumes, note velocities and expression will be scaled.</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QSpinBox" name="midiExAdlChips">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="toolTip">
              <string>The number of emulated OPL3 chips. Increase the count of chips to get wider polyphony (18 of two-operator voices per chip, or 6 four-operator voices per chip), or decrease it if you have the choppy music playback.</string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>4</number>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QCheckBox" name="midiExAdlChipsEn">
             <property name="text">
              <string>Chips number:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="4" column="0" colspan="4">
         <widget class="QGroupBox" name="midiSetupOPN">
          <property name="title">
           <string>Extra settings</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5" columnstretch="0,0,0">
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <item row="0" column="2">
            <widget class="QPushButton" name="pushButton_2">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Browse...</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1" colspan="2">
            <widget class="QComboBox" name="midiExOpnVolumeModel">
             <property name="toolTip">
              <string>Volume model declares how music volumes, note velocities and expression will be scaled.</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="lineEdit_2">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string notr="true">&lt;WIP&gt;</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QSpinBox" name="midiExOpnChips">
             <property name="toolTip">
              <string>The number of emulated OPL3 chips. Increase the count of chips to get wider polyphony (18 of two-operator voices per chip, or 6 four-operator voices per chip), or decrease it if you have the choppy music playback.</string>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>100</number>
             </property>
             <property name="value">
              <number>8</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>Volume model:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_10">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Custom bank:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QCheckBox" name="midiExOpnChipsEn">
             <property name="text">
              <string>Chips number:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="5" column="1">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>2</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabChiptune">
       <attribute name="title">
        <string>Chiptune</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item row="3" column="5">
         <widget class="QPushButton" name="gmeGainReset">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
        <item row="3" column="3" colspan="2">
         <widget class="QDoubleSpinBox" name="gmeGainAbs">
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="value">
           <double>2.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="3" colspan="2">
         <widget class="QDoubleSpinBox" name="gmeTempoAbs">
          <property name="minimum">
           <double>0.010000000000000</double>
          </property>
          <property name="maximum">
           <double>100.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="5">
         <widget class="QPushButton" name="gmeTempoReset">
          <property name="text">
           <string>Reset</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Tempo:</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1" colspan="2">
         <widget class="QSlider" name="gmeGain">
          <property name="maximum">
           <number>400</number>
          </property>
          <property name="value">
           <number>200</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksAbove</enum>
          </property>
          <property name="tickInterval">
           <number>50</number>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Gain:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1" colspan="2">
         <widget class="QSlider" name="gmeTempo">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>200</number>
          </property>
          <property name="value">
           <number>100</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="tickPosition">
           <enum>QSlider::TicksAbove</enum>
          </property>
          <property name="tickInterval">
           <number>25</number>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>261</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0" colspan="6">
         <widget class="QGroupBox" name="groupBox">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>38</height>
           </size>
          </property>
          <property name="title">
           <string>Track number</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <item row="1" column="0">
            <widget class="QPushButton" name="gmeGoToFirst">
             <property name="text">
              <string>To begin</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="gmeTrackNumber">
             <property name="maximum">
              <number>99999</number>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QPushButton" name="gmePrevTrack">
             <property name="text">
              <string>Previous</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="gmeNextTrack">
             <property name="text">
              <string>Next</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QLabel" name="label_12">
      <property name="text">
       <string>Tip: to preview the settings result, enable music playing, please. (Look for the play/stop note icon on the main window toolbar).</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>midiExAdlChipsEn</sender>
   <signal>toggled(bool)</signal>
   <receiver>midiExAdlChips</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>303</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>303</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>midiExOpnChipsEn</sender>
   <signal>toggled(bool)</signal>
   <receiver>midiExOpnChips</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>119</x>
     <y>487</y>
    </hint>
    <hint type="destinationlabel">
     <x>196</x>
     <y>494</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
